docs/tech_notes/2025-2-21-CN-telemetry-setup-guide.md

```diff
@@ -0,0 +1,131 @@
+instructions [here](https://github.com/GaloisInc/VERSE-Toolchain/blob/main/cn-client/README.md#installing). 
+
+Here’s what I’d expect to see: 
+
+```sh
+$ cd $TELEMETRY/VERSE-Toolchain/cn-client   # go to cn-client directory 
+$ npm install
+$ npm run compile
+$ npm run dist
+$ code --install-extension cn-client-0.0.5.vsix   # The version number may change 
+```
+
+You should now see a CN plugin in the VS code extensions menu (on the left-hand
+side of the editor, or View-\>Extensions in the menu). 
+
+## *Step 3:* Install the CN LSP server and telemetry collection 
+
+To install the CN LSP server and telemetry collection tool, follow the
+instructions [here](https://github.com/GaloisInc/VERSE-Toolchain/blob/main/README.md#installation-and-use). 
+
+Here’s what I’d expect to see (assuming you’ve installed OCaml and opam): 
+
+```sh
+$ cd $TELEMETRY/VERSE-Toolchain   # go to root of VERSE-Toolchain repository
+$ opam update
+$ opam switch create . ocaml.5.1.1 --locked -y
+$ eval $(opam env)
+```
+
+## *Step 4:* Create the telemetry data directory 
+
+Telemetry data is stored in a single fixed directory, regardless which file is
+being analyzed with CN. 
+
+```sh
+$ mkdir $TELEMETRY/CN-telemetry-data 
+```
+
+## *Step 5:* Set up the VS Code client 
+
+We need to tell the telemetry where to store data. Open the CN plugin by
+clicking on the Extensions menu, or through View-\>Extensions. Then open the
+‘Settings’ menu (click on the cog, then ‘Settings’). Now update the settings as
+follows (make sure to substitute your `TELEMETRY` root directory): 
+
+* CN: Telemetry Dir - `<TELEMETRY>/CN-telemetry-data`  
+* CN: User ID - *your email address* 
+
+We also need to tell VS Code where to find the CN runtime files. You should be
+prompted to pick the opam switch which you installed in Step 3 - see
+[here](https://github.com/GaloisInc/VERSE-Toolchain/tree/main/cn-client#running-cn)
+for details. In our case, you should pick the switch
+`<TELEMETRY>/VERSE-Toolchain/`. If this doesn’t work for some reason, you can
+also set the location of the CN files manually in the same way as the telemetry
+data. Here are the settings you’ll need: 
+
+* CN: Cerb Runtime - `<TELEMETRY>/VERSE-Toolchain/_opam/lib/cerberus/runtime`  
+* CN: Server Path - `<TELEMETRY>/VERSE-Toolchain/_opam/bin/cn-lsp-server`
+
+## *Step 6:* Test everything 
+
+Let’s try this all out on an example. Here’s a piece of C with CN annotations
+(from the tutorial, [here](https://rems-project.github.io/cn-tutorial/getting-started/tutorials/basic-usage/#first-function-specification)). 
+
+```c
+int add(int x, int y)
+/*@ requires let Sum = (i64) x + (i64) y;
+             -2147483648i64 <= Sum; Sum <= 2147483647i64; @*/
+{
+  return x+y;
+}
+```
+
+Copy this code into a file. VS Code should add syntax highlighting to the CN
+annotations - the part inside ‘magic comments’, `/*@ ... @*/`. You should also see
+a button above the `add()` function that says ‘Verify with CN’. When you click
```

Comment:
I changed these buttons so that they include the name of the function they pertain to, so it would be more accurate to say:
```suggestion
a button above the `add()` function that says ‘Verify add with CN’. When you click
```